### PR TITLE
VS-361 Add GvsWithdrawSamples wdl

### PR DESCRIPTION
**Pull request by gbggrant**

### CODE REVIEW

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+    # perform actual update
+    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false \
+      'UPDATE `~{dataset_name}.~{sample_info_table}` SET withdrawn = CURRENT_TIMESTAMP() WHERE sample_name IN ("~{sep='\", \"' sample_names}")' > log_message.txt;
+
+    cat log_message.txt | sed -e 's/Number of affected rows: //' > rows_updated.txt
+    typeset -i rows_updated=$(cat rows_updated.txt)
+
+    echo "did not update $num_samples rows - only updated $rows_updated"
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:79]: seems like this should only be executed if the counts are mismatched (like the other copy of this on line 82)

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    if [ $num_samples -ne $rows_updated ]; then
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:81]: I really like having this check...I wish there was a cleaner way of doing this with the `bq` command line tool but if there is I haven't found it... 😞

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+task WithdrawSamples {
+  input {
+    String project_id
+    String dataset_name
+
+    String sample_info_table
+    Array[String] sample_names
+
+    # runtime
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:40]: Looking at the other WDLs in this repo this comment appears to be marking inputs that are used in expressions in the `runtime` section of tasks. Assuming I have that right (which I might not), `# runtime` would not apply to these two variables.

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    set -x
+
+    # make sure that sample names were actually passed, warn and exit if empty
+    num_samples=~{length(sample_names)}
+    if [ $num_samples -eq 0 ]; then
+      echo "No sample names passed. Exiting"
+      exit 0
+    fi
+
+    export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:63]: It seems `gatk_override` isn't actually needed in this WDL and could be removed (there are no GATK invocations here).

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl (hunk 1 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/GvsWithdrawSamples.wdl:79]: Yeah, that's some leftover debugging code.

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl (hunk 3 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/GvsWithdrawSamples.wdl:40]: Yeah, that's true. I'll remove the comment.

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl (hunk 4 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/GvsWithdrawSamples.wdl:63]: Thanks.

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  input {
+    String dataset_name
+    String project_id
+
+    Array[String] sample_names
+
+    String? service_account_json_path
+  }
+
+  String sample_info_table = "sample_info"
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:14]: Is there a use case that benefits from `sample_info_table` being a declaration here (and an input in `WithdrawSamples`) or could its usage just be hardcoded `sample_info` in `WithdrawSamples`?

File: scripts/variantstore/wdl/GvsWithdrawSamples.wdl [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    if [ $num_samples -ne $rows_updated ]; then
+      echo "Error: Expected to update $num_samples rows - but only updated $rows_updated."
+      exit 1
+    fi
+
+  >>>
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.7.0"
+    memory: "3.75 GB"
+    disks: "local-disk " + 10 + " HDD"
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsWithdrawSamples.wdl:82]: can this be just one string rather than a concatenation of literals?

Comment (gbggrant) [reply scripts/variantstore/wdl/GvsWithdrawSamples.wdl:82]: Yeah - would have been a useful pattern if there was a variable for 10

**Review by mcovarr — APPROVED** [review]

**Review by rsasch — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**